src/tree/registries/IRegistryProvider.ts

```diff
@@ -54,5 +55,10 @@ export interface IRegistryProvider {
     /**
      * The tree item class to instantiate after a provider is connected
      */
-    treeItemType: new (parent: AzExtParentTreeItem, cached: ICachedRegistryProvider) => AzExtParentTreeItem & IRegistryProviderTreeItem;
+    treeItemType: new (parent: AzExtParentTreeItem, provider: IRegistryProvider, cachedProvider: ICachedRegistryProvider) => AzExtParentTreeItem & IRegistryProviderTreeItem;
```

Comment:
It seems like it would've been easier had `treeItemType` been a factory function rather than a type; that would have allowed each registry provider to instantiate whatever object it wanted, with whatever arguments were needed (e.g. an auth provider), without having to make those dependencies known to *all* registry tree item types (e.g. those that don't need any auth).